src/llama-sampling.cpp

```diff
@@ -1226,7 +1226,9 @@ static struct llama_sampler_i llama_sampler_penalties_i = {
     /* .name   = */ [](const struct llama_sampler * /*smpl*/) { return "penalties"; },
     /* .accept = */ [](struct llama_sampler * smpl, llama_token token) {
         auto * ctx = (llama_sampler_penalties *) smpl->ctx;
-        ctx->prev.push_back(token);
+        if (ctx->prev.size()) {
+            ctx->prev.push_back(token);
+        }
```

Comment:
I think this is not correct, it should check the capacity rather than the size (which will always be zero unless something is pushed). I will push a fix later, but at the moment this sampler is probably broken.

Reply:
Ah yes. It should check the capacity